src/core/fileutils.h

```diff
@@ -0,0 +1,24 @@
+#ifndef FILEUTILS_H
+#define FILEUTILS_H
+
+#include <QObject>
+
+class FileUtils : public QObject
+{
+    Q_OBJECT
+
+  public:
+    FileUtils( QObject *parent = nullptr );
+    //! Destructor
+    ~FileUtils() = default;
+
+    //! returns the mimetype of a filepath as string
+    Q_INVOKABLE static QString mimeTypeName( const QString filePath );
+    //! returns the filename of a filepath - if no file name exists it's empty
+    Q_INVOKABLE static QString fileName( const QString filePath );
```

Comment:
```suggestion
    Q_INVOKABLE static QString fileName( const QString& filePath );
```

Also on the other two methods